azkaban-exec-server/src/main/bash/azkaban-executor-shutdown.sh

```diff
@@ -5,15 +5,30 @@ installdir="$(dirname $0)/.."
 pid=`cat ${installdir}/currentpid`
 port=`cat ${installdir}/executor.port`
 
-echo "Killing Executor. [pid: $pid, port: $port]"
-
-kill ${pid}
-if [ $? -ne 0 ]; then
-    echo "Error: Shutdown failed"
-    exit 1;
+maxtry=5
+if [ -z $pid ]; then
```

Comment:
Could you add 'set -o nounset' guard against referencing undefined variables?  You will also want to add 'set -o errexit' to force an exit if any shell-outs  return a non-zero exit.  For example, if the cat command fails to find the currentpid file, the script will continue running and $pid is unset which is referenced on line 9.  By using both 'set -o' options, we can guard against this ocuring by forcing an early failure.